src/main/java/tdl/record/sourcecode/ListCommand.java

```diff
@@ -21,25 +22,25 @@
     public void run() {
         File file = Paths.get(inputFilePath).toFile();
         try (SnapshotsFileReader reader = new SnapshotsFileReader(file)) {
+            Date start = new Date(reader.getFileHeader().getTimestamp());
             int index = 0;
             while (reader.hasNext()) {
                 SnapshotFileSegment segment = reader.next();
-                printSnapshot(segment, index);
+                printSnapshot(segment, index, start);
                 index++;
             }
         } catch (IOException ex) {
             throw new RuntimeException(ex);
         }
     }
 
-    private static void printSnapshot(SnapshotFileSegment segment, int index) {
+    private void printSnapshot(SnapshotFileSegment segment, int index, Date start) {
         String type = segment.getSnapshot() instanceof KeySnapshot ? "KEY" : "PATCH";
         long size = segment.size + SnapshotFileSegment.HEADER_SIZE;
         String checksum = Hex.encodeHexString(segment.checksum);
-
-        String infoLine = String.format("#%4d | time %4s | type %-5s | offset %5d | size %5d | checksum %40s",
-                index, segment.timestamp, type, segment.address, size, checksum);
+        String recorded = (new Date(start.getTime() + segment.timestamp)).toString();
+        String infoLine = String.format("#%4d | recorded %40s | time %4s | type %-5s | offset %5d | size %5d | checksum %40s",
```

Comment:
The `start.getTime()` should only be displayed once, before the snapshots.
For each snapshot it is easier if we only display `segment.timestamp`.